superset/assets/src/dashboard/actions/dashboardLayout.js

```diff
@@ -163,10 +163,30 @@ export function handleComponentDrop(dropResult) {
     const droppedOnRoot = destination && destination.id === DASHBOARD_ROOT_ID;
     const isNewComponent = source.id === NEW_COMPONENTS_SOURCE_ID;
 
+    const { dashboardLayout: undoableLayout } = getState();
+    const { present: layout } = undoableLayout;
+    const dashboardRoot = layout[DASHBOARD_ROOT_ID];
+    const rootChildId =
+      dashboardRoot && dashboardRoot.children ? dashboardRoot.children[0] : '';
+
     if (droppedOnRoot) {
       dispatch(createTopLevelTabs(dropResult));
     } else if (destination && isNewComponent) {
       dispatch(createComponent(dropResult));
+    } else if (
+      // Add additional allow-to-drop logic for tag/tags source.
+      // we only allow
+      // - top-level tab => top-level tab: rearrange top-level tab order
+      // - nested tab => top-level tab: allow row tab become top-level tab
+      // we do not allow top-level tab become nested tab
+      source.type === TABS_TYPE &&
+      destination.type === TABS_TYPE &&
+      source.id === rootChildId &&
+      destination.id !== rootChildId
+    ) {
+      return dispatch(
+        addInfoToast(`Can not move top level tab into nested tabs`),
```

Comment:
Should this be a `t()` call?

Reply:
Also, should this be a `warning` toast since it's an invalid action?

Reply:
added `t` function. I saw other message, `There is not enough space for this component` is also info not warning. should i can make both of them warning message?

Reply:
Up to you, I think consistency is good so either both `info` or both `danger` sounds good to me 👍

Reply:
i updated both message to `danger`. Will merge after CI passed.